### DISCUSSION TITLE
What can i define to walk into behind a object?

### DISCUSSION THREAD

**Discussion by ndkk** [flareteam/flare-game#943] [Q&A] [answered]
I walk on top of a house. 

I want to hidden in this house.

How can i do it ? 

as below:

[https://github.com/ndkk/uploadfile/blob/main/onroof.png](https://github.com/ndkk/uploadfile/blob/main/onroof.png)

**Comment by dorkster** [comment]
First, you want to make sure that the tiles for the house are not placed on the bottom-most layer. For our maps, we will typically put all the floor tiles in a layer called "background" and everything else in a layer called "object".

Next, your house tiles should be shaped something like this (highlighted yellow):
![tiles1](https://github.com/flareteam/flare-game/assets/913982/b9e7a156-3d19-4cf9-bf99-acdeebff5210)
So based on the 44 x 44 pixel floor tile size in your screenshot, your object tiles should measure 44 x 176. Objects are positioned at the bottom of the tile, with blank space above depending on how tall the object is. So the barrel, for example, would only use the bottom quarter of a single tile.

You'll also want to set up a "collision" tile layer to prevent the player from walking through the house. Here's an example of what that might look like:
![tiles2](https://github.com/flareteam/flare-game/assets/913982/9cc012bd-9222-4639-a8a7-7a4003d5eeda)
Collision tiles should be the same size as your floor tiles (44 x 44 pixels).

**Reply by ndkk** [reply]
Thanks for the detailed clarification. 
But the above I have already done, it couldn't work.

I uploaded all the mod and tiled into my git repository.
the scene in city.tmx as below:
[https://github.com/ndkk/hero/blob/main/tiled/geos/city.tmx](https://github.com/ndkk/hero/blob/main/tiled/geos/city.tmx)

Could you help me find the correct config ?

**Reply by dorkster** [reply, marked as answer]
The problem is that the house tiles are still 32x32 square tiles instead of 32x128 "strips". To demonstrate, I'll create a new tileset with just the house and the signs:

![maps_tall](https://github.com/flareteam/flare-game/assets/913982/8f7e3e33-afad-4f6d-9503-48fc18c667bd)
![Screenshot_20230615_125149](https://github.com/flareteam/flare-game/assets/913982/ef7a487e-16b0-40c1-8664-9e59ece40c73)
![Screenshot_20230615_125301](https://github.com/flareteam/flare-game/assets/913982/95218d9b-1cff-4a1c-9e0c-c13084591399)

As you can see, each tile is a tall rectangle instead of a square.

I'll then add the new tileset to the tileset definition `city_dst.txt`. Make note of the Y offset of 112 pixels. That places the "origin" point of the tile at the center of the bottom segment.

```
[tileset]
img=images/tilesets/maps_tall.png
tile=1001,0,0,32,128,16,112
tile=1002,32,0,32,128,16,112
tile=1003,64,0,32,128,16,112
tile=1004,96,0,32,128,16,112
tile=1005,128,0,32,128,16,112
tile=1006,160,0,32,128,16,112
tile=1007,192,0,32,128,16,112
tile=1008,224,0,32,128,16,112
```

And here's the in-game result:
![Screenshot_20230615_130723](https://github.com/flareteam/flare-game/assets/913982/9faae7d3-62ae-43aa-9157-180d16e56062)

**Reply by ndkk** [reply]
Great! It worked ! Thanks so much !  :+1:  :smile:
